src/poetry/utils/password_manager.py

```diff
@@ -170,12 +170,22 @@ def set_pypi_token(self, name: str, token: str) -> None:
         else:
             self.keyring.set_password(name, "__token__", token)
 
-    def get_pypi_token(self, name: str) -> str | None:
-        if not self.keyring.is_available():
-            token: str | None = self._config.get(f"pypi-token.{name}")
-            return token
-
-        return self.keyring.get_password(name, "__token__")
+    def get_pypi_token(self, repo_name: str) -> str | None:
+        """Get PyPi token.
+
+        First checks the environment variables for a token,
+        then the configured username/password and the
+        available keyring.
+
+        :param repo_name:  Name of repository.
+        :return: Returns a token as a string if found, otherwise None.
+        """
+        return self._config.get(f"pypi-token.{repo_name}", self.keyring.get_password(repo_name, "__token__"))
+        return (
```

Comment:
two returns? i guess first return is a token variable

Reply:
fixed